micro-rdk/src/esp32/pwm.rs

```diff
@@ -219,14 +219,31 @@ impl<'a> LedcTimerWrapper<'a> {
         if self.count > 0 {
             return Err(Esp32PwmError::OtherChannelsBindToTimer);
         }
+
         let id = {
             let timer = self.timer.take();
             timer.unwrap().timer() as u8
         };
         let timer_config = TimerConfig::default().frequency(frequency_hz.Hz());
-        let _ = self.timer.set(create_timer_driver(id, &timer_config)?);
-        self.frequency = frequency_hz;
-        Ok(())
+
+        match create_timer_driver(id, &timer_config) {
+            Ok(driver) => {
+                let _ = self.timer.set(driver);
+                self.frequency = frequency_hz;
+                Ok(())
+            }
+            Err(err) => {
+                // if there is a failure, or the frequency is bad, we want to revert back
+                // to the last working frequency before raising the error
+                let timer_config = TimerConfig::default().frequency(self.frequency.Hz());
+                let _ =
+                    self.timer
+                        .set(create_timer_driver(id, &timer_config).unwrap_or_else(|_| {
```

Comment:
nit: replace with an expect here

Reply:
I tried `.expect(format!(...))` but clippy got mad at me and insisted I do it this way

Reply:
damm clippy :P